lib/MetaCPAN/Query/Release.pm

```diff
@@ -1176,10 +1189,17 @@ sub find_download_url {
         );
     }
 
-    return +{
-        %{ $hit->{_source} },
-        %{ $hit->{inner_hits}{module}{hits}{hits}[0]{_source} }, @checksums,
-    };
+    my $source_name = delete $source->{name};
+    if ( !$module_filter ) {
+        $source->{release} = $source_name;
+    }
+
+    my $module
+        = $hit->{inner_hits}{module}
+        ? $hit->{inner_hits}{module}{hits}{hits}[0]{_source}
+        : {};
```

Comment:
probably always want to add a `$module //= {}` as we are going to dereference it later

Reply:
The ES query that would produce `{inner_hits}{module}` requires a module with a matching name. If `inner_hits` exists, `{inner_hits}{module}{hits}{hits}[0]{_source}` will always exist.